### ISSUE TITLE
Tests not running on Core 3.1

### ISSUE THREAD

**Issue by adam-becker** [YoloDev/YoloDev.Expecto.TestSdk#28]
I am getting garbled output when running tests with this adapter using `dotnet test`.

Before pressing `^C` the output just stalls.

```# dotnet test
Test run for /workspace/Nemo.Platform.Tests/bin/Debug/netcoreapp3.1/Nemo.Platform.Tests.dll(.NETCoreApp,Version=v3.1)
Test run for /workspace/Nemo.Processors.RP65816.Tests/bin/Debug/netcoreapp3.1/Nemo.Processors.RP65816.Tests.dll(.NETCoreApp,Version=v3.1)
Microsoft (R) Test Execution Command Line Tool Version 16.3.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Microsoft (R) Test Execution Command Line Tool Version 16.3.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
Starting test execution, please wait...
Test run for /workspace/Nemo.Processors.ARM7.Tests/bin/Debug/netcoreapp3.1/Nemo.Processors.ARM7.Tests.dll(.NETCoreApp,Version=v3.1)
Test run for /workspace/Nemo.Processors.LR35902.Tests/bin/Debug/netcoreapp3.1/Nemo.Processors.LR35902.Tests.dll(.NETCoreApp,Version=v3.1)

A total of 1 test files matched the specified pattern.

A total of 1 test files matched the specified pattern.
Test run for /workspace/Nemo.Processors.RP6502.Tests/bin/Debug/netcoreapp3.1/Nemo.Processors.RP6502.Tests.dll(.NETCoreApp,Version=v3.1)
Microsoft (R) Test Execution Command Line Tool Version 16.3.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Microsoft (R) Test Execution Command Line Tool Version 16.3.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Microsoft (R) Test Execution Command Line Tool Version 16.3.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Test run for /workspace/Nemo.GameBoy.Tests/bin/Debug/netcoreapp3.1/Nemo.GameBoy.Tests.dll(.NETCoreApp,Version=v3.1)
Starting test execution, please wait...
Starting test execution, please wait...

A total of 1 test files matched the specified pattern.

A total of 1 test files matched the specified pattern.
Starting test execution, please wait...
Microsoft (R) Test Execution Command Line Tool Version 16.3.0
                                                                                                                                                              
Additionally, path to test adapters can be specified using /TestAdapterPath command. Example  /TestAdapterPath:<pathToCustomAdapters>.
Test run for /workspace/Nemo.SuperFamicom.Tests/bin/Debug/netcoreapp3.1/Nemo.SuperFamicom.Tests.dll(.NETCoreApp,Version=v3.1)
Microsoft (R) Test Execution Command Line Tool Version 16.3.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...cified pattern.
                                                                                                                                                              A total of 1 test files matched the specified pattern./bin/Debug/netcoreapp3.1/Nemo.Platform.Tests.dll. Make sure that test discoverer & executors are registeAttempting to cancel the build...ion settings are appropriate and try again.
Attempting to cancel the build...         6 |
Attempting to cancel the build...
Attempting to cancel the build...
Attempting to cancel the build...
Attempting to cancel the build...
Attempting to cancel the build...
Attempting to cancel the build...```

**Comment by adam-becker** [comment]
Also, running `dotnet test --list-tests` produces some partially correct output before choking:

```$ dotnet test --list-tests
Test run for /workspace/Nemo.Platform.Tests/bin/Debug/netcoreapp3.1/Nemo.Platform.Tests.dll(.NETCoreApp,Version=v3.1)
Test run for /workspace/Nemo.Processors.LR35902.Tests/bin/Debug/netcoreapp3.1/Nemo.Processors.LR35902.Tests.dll(.NETCoreApp,Version=v3.1)
Microsoft (R) Test Execution Command Line Tool Version 16.3.0
Copyright (c) Microsoft Corporation.  All rights reserved.

The following Tests are available:
Test run for /workspace/Nemo.Processors.ARM7.Tests/bin/Debug/netcoreapp3.1/Nemo.Processors.ARM7.Tests.dll(.NETCoreApp,Version=v3.1)
Microsoft (R) Test Execution Command Line Tool Version 16.3.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Test run for /workspace/Nemo.Processors.RP65816.Tests/bin/Debug/netcoreapp3.1/Nemo.Processors.RP65816.Tests.dll(.NETCoreApp,Version=v3.1)
Microsoft (R) Test Execution Command Line Tool Version 16.3.0
Copyright (c) Microsoft Corporation.  All rights reserved.

The following Tests are available:
Test run for /workspace/Nemo.SuperFamicom.Tests/bin/Debug/netcoreapp3.1/Nemo.SuperFamicom.Tests.dll(.NETCoreApp,Version=v3.1)
Microsoft (R) Test Execution Command Line Tool Version 16.3.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Test run for /workspace/Nemo.Processors.RP6502.Tests/bin/Debug/netcoreapp3.1/Nemo.Processors.RP6502.Tests.dll(.NETCoreApp,Version=v3.1)
The following Tests are available:
The following Tests are available:
Microsoft (R) Test Execution Command Line Tool Version 16.3.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Microsoft (R) Test Execution Command Line Tool Version 16.3.0
Copyright (c) Microsoft Corporation.  All rights reserved.

The following Tests are available:
The following Tests are available:
^[[11;1R    addressing/absolute address/wraps within current bank
    addressing/absolute address x/wraps within current bank
    addressing/absolute address x/carries to next bank when adding x
    addressing/absolute address, long/wraps within current bank
    addressing/absolute address x, long/wraps within current bank
    addressing/absolute address x, long/carries to next bank when adding x
^CAttempting to cancel the build...
Attempting to cancel the build...
Attempting to cancel the build...
Attempting to cancel the build...
Attempting to cancel the build...
Attempting to cancel the build...
Attempting to cancel the build...
Attempting to cancel the build...```

**Comment by Alxandr** [comment]
Can you add the output form `dotnet --info`?

**Comment by adam-becker** [comment]
```$ docker run --rm -it mcr.microsoft.com/dotnet/core/sdk:3.1 dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   3.1.200
 Commit:    c5123d973b

Runtime Environment:
 OS Name:     debian
 OS Version:  10
 OS Platform: Linux
 RID:         debian.10-x64
 Base Path:   /usr/share/dotnet/sdk/3.1.200/

Host (useful for support):
  Version: 3.1.2
  Commit:  916b5cba26

.NET Core SDKs installed:
  3.1.200 [/usr/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.1.2 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.2 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download```

**Comment by Alxandr** [comment]
Is this running on a CI, or on a local machine? I've been trying to increase the test infrastructure in this repository, and I've come across an error if you *only* have .net core 3.x installed. Can you try to install both 2.x (latest) and whatever 3.x version you require and see if that solves your issue? In the meantime, I'll try to figure out how to fix the underlying issue.

**Comment by adam-becker** [comment]
This is running on the official docker image (`mcr.microsoft.com/dotnet/core/sdk:3.1`). It looks like the console is getting output asynchronously, so you see the expected half-garbled output. Then everything locks up until you send `SIGINT` (`ctrl+c`) to kill the process.

**Comment by Alxandr** [comment]
Hmm. I'm unsure about that. I'm unable to reproduce the issue myself, so I'm not entirely sure how to move forward.

**Comment by adam-becker** [comment]
So, I retried with whatever the latest version is, and now the tests run to completion but the output is still garbled. Looks like the version bumped, from `3.1.200` to `3.1.201`.

```$ dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   3.1.201
 Commit:    b1768b4ae7

Runtime Environment:
 OS Name:     debian
 OS Version:  10
 OS Platform: Linux
 RID:         debian.10-x64
 Base Path:   /usr/share/dotnet/sdk/3.1.201/

Host (useful for support):
  Version: 3.1.3
  Commit:  4a9f85e9f8

.NET Core SDKs installed:
  3.1.201 [/usr/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.1.3 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.3 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download```

**Comment by Alxandr** [comment]
What do you mean the output is garbled?

**Comment by adam-becker** [comment]
If there's 5 test projects in the same solution, the output on stdio is a mix of all 5 processes running at once. It's all interleaved.

**Comment by Alxandr** [comment]
I don't think that has anything to do with YoloDev.Expecto.TestSdk, that's just VSTest running the tests in parallel as far as I know.

**Comment by adam-becker** [comment]
Alright, looks like this was most likely caused by a bug that was patched between version `3.1.200` and `3.1.201` of DNC. Thanks for following up @Alxandr
